pkg/testing/match/match.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package match
+
+import (
+	"encoding/json"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func parseMultiJSONOutput[T any](t *testing.T, output string, normalize func(*T)) []*T {
```

Comment:
You could remove the functions that appear in this file from `integration/helpers.go` and include this package there. Would avoid code duplication.
Best would be a separate commit for this before the other changes

Reply:
Maybe I should change these functions in a different PR? (mostly the built-in to image-based gadgets test conversion PR).

Reply:
I think we can do it right away on this PR.